File: app/src/main/java/com/example/record/model/YearlyData.kt
package com.example.record.model

data class YearlyData (
    val year: Int,
    val recordMap: List<CategoryRecord>,
    val totalIncome: Float,
    val totalOutcome: Float
)

data class CategoryRecord(
    val category: String,
    val totalIncome: Float,
    val totalOutcome: Float,
    val items: List<SubCategoryRecord>
)

data class SubCategoryRecord(
    val subCategory: String,
    val totalIncome: Float,
    val totalOutcome: Float,
    val items: List<Record>
)

File: Record/Model/YearlyData.swift
//
//  YearlyData.swift
//  Record
//
//  Created by xiachuan on 2024/9/1.
//

import Foundation

struct YearlyData {
    var year: Int
    var totalIncome: Float
    var totalOutcome: Float
    var recordMap: [CategoryRecord]
}

struct CategoryRecord: Identifiable {
    let id: UUID = UUID()
    var category: String
    var totalIncome: Float
    var totalOutcome: Float
    var items: [SubCategoryRecord]
}

struct SubCategoryRecord: Identifiable {
    let id: UUID = UUID()
    var subCategory: String
    var totalIncome: Float
    var totalOutcome: Float
    var items: [Record]
}
